pennylane/labs/vibrational/christiansen_ham.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The functions related to the construction of the Christiansen form Hamiltonian."""
+
+import numpy as np
+
+# pylint: disable = too-many-branches, too-many-positional-arguments
+
+from pennylane.bose import BoseSentence, BoseWord, christiansen_mapping
+from .christiansen_utils import christiansen_integrals, christiansen_integrals_dipole
+
+
+def christiansen_bosonic(one, modes=None, modals=None, two=None, three=None, ordered=True):
+    r"""Build the bosonic operators in the Christiansen form.
+
+    Args:
+        one (array): 3D array with one-body matrix elements
+        modes (int): number of vibrational modes, detects from 'one' if none is provided
+        modals (array): 1D array with the number of allowed vibrational modals for each mode, detects from 'one' if none is provided
+        two (array): 6D array with two-body matrix elements
+        three (array): 9D array with three-body matrix elements
+        cutoff (float): magnitude beneath which terms are not incorporated in final expression
+        ordered (bool): Flag for if matrix elements are already ordered. Default is True
```

Comment:
Please provide the correct return-types and maybe we can use better variable names for one, two and three.